data_pipelines_cli/cli_commands/compile.py

```diff
@@ -28,14 +30,33 @@
 from ..io_utils import replace
 
 
+def _read_dbt_vars_from_config(dbt_env_config: Dict[str, Any]) -> str:
+    dbt_vars: Dict[str, str] = dbt_env_config.get("vars", {})
+    return yaml.dump(dbt_vars, default_flow_style=True, width=sys.maxsize)
+
+
 def dbt(
     command: Tuple[str, ...], env: str, profiles_path: Optional[pathlib.Path]
 ) -> None:
     command_str = " ".join(list(command))
     echo_subinfo(f"dbt {command_str}")
+
     profiles_path = profiles_path or generate_profiles_yml(env)
+    dbt_env_config = read_config(BUILD_DIR.joinpath("dag"), env, "dbt.yml")
+    dbt_vars = _read_dbt_vars_from_config(dbt_env_config)
     subprocess_run(
-        ["dbt", *command, "--profiles-dir", str(profiles_path), "--target", env]
+        [
+            "dbt",
+            *command,
+            "--profile",
+            dbt_env_config["target_type"],
```

Comment:
This fixes a bug that went unnoticed: `dbt` searches for `profile: <YOUR_PROFILE>` field in `dbt_project.yml` and then tries to get a dictionary referenced by the `<YOUR_PROFILE>` key in `profiles.yml`. However, `profiles.yml` gets generated with `target_type` as this "root key".